### PR TITLE
fix(web): support saving focus for custom OSK interactions

### LINKED ISSUE #5944: bug(web): sil_euro_latin osk disappears on 2nd click
**Describe the bug**

On a desktop browser, clicking on any "key" in the sil_euro_latin OSK causes the OSK to disappear -- but only on the 2nd click. The first click works fine.

image

This issue is present in 13.0-stable through 15.0-alpha. I have not tested earlier versions. It does not seem to repro on keymanweb.com but does repro on http://localhost/keymanweb/testing/unminified.html. This suggests there may be interactions between the OSK and the keyboard selector UI?

I thought at first that it was related to an issue within the `setFocusTimer()` function that is referenced in #5909, but the issue arises whether or not the `setFocusTimer()` functionality is present.

I resolved the error arising in #5939 locally and that did not address this problem.

I note that the `keymanweb.KSF()` (aka `saveFocus()`) function has no effect. This is probably the root cause, and I am guessing it is probably related to earlier refactors.

**To Reproduce**

1. Go to http://localhost/keymanweb/testing/unminified.html
2. Select Swedish
3. Click on a letter "key" on the sil_euro_latin OSK.
4. Click on the same letter key or any other letter key again.
5. Observe the OSK disappearing (as focus is lost)

**Expected behavior**

Well, the OSK shouldn't disappear...

---

**KeymanWeb:**
 - KeymanWeb Version: observed in 13.0-stable - 15.0-alpha

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #5944.

If a custom OSK, such as sil_euro_latin, calls `keymanweb.KSF()` or `keymanweb.saveFocus()`, this ensures that the next focus change is ignored, which prevents the OSK from being dismissed as soon as it is clicked on.

This is a regression, probably arising in version 10.0.

Note: `_SelectionChange` is no longer used or referenced, so I removed it as part of this fix.

# User Testing

* TEST_EUROLATIN: Load unminified test on a desktop browser, and select Swedish (Eurolatin). Ensure that clicking on "keys" in the custom Eurolatin OSK does not dismiss the OSK or lose focus. Experiment with switching between input fields, and ensure that the OSK continues to function correctly even after switching fields. Ensure that the OSK disappears when you click on a blank area of the page.
* TEST_BASELINE: Switch to another keyboard layout and check that the OSK continues to function correctly and be hidden correctly when you click outside a text area. Switch back and forth between this keyboard and eurolatin and ensure that the OSK continues to function correctly.

### CODE REVIEW

File: web/source/dom/domEventHandlers.ts [hunk 1]
```diff
@@ -197,6 +197,15 @@ namespace com.keyman.dom {
         }
       }
 
+      if(DOMEventHandlers.states._IgnoreNextSelChange) {
+        // If a keyboard calls saveFocus() (KSF), then ignore the
```
Comment (darcywong00) [inline web/source/dom/domEventHandlers.ts:201]: Where's the check for `KSF`?

Comment (mcdurdin) [reply web/source/dom/domEventHandlers.ts:201]: `_IgnoreNextSelChange` is the value set by `KSF()`/`saveFocus()`.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by MakaraSok** [comment]
* **TEST_EUROLATIN: FAILED?** The OSK behaves as expected on Chrome, FireFox and Safari on macOS and FireFox on Windows; except Chrome on Windows where the OSK appears narrow and it cannot be switch to the EuroLatin keyboard.


   https://user-images.githubusercontent.com/28331388/143158903-9edb0b8f-3986-457d-825a-f177f33587fc.mov

* **TEST_BASELINE: FAILED?** The OSK behaves as expected o Chrome, FireFox and Safari on macOS and FireFox on Windows; except Chrome on Windows where the OSK appears narrow and cannot be switched to Lao keyboard successfully.

   image

**Comment by mcdurdin** [comment]
> * except Chrome on Windows where the OSK appears narrow and it cannot be switch to the EuroLatin keyboard.

Did you get any script errors in the console?

**Comment by MakaraSok** [comment]
* **TEST_EUROLATIN: PASSED** The keyboard is now working as expected on Chrome in Windows. 

* **TEST_BASELINE: PASSED** It shows as normal, switch as normal and hide/show as normal.

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.157-alpha](https://keyman.com/downloads/releases/alpha/15.0.157)

**Pull request merged**